gazebo_plugins/src/gazebo_ros_camera.cpp

```diff
@@ -161,23 +174,48 @@ void GazeboRosCamera::Load(gazebo::sensors::SensorPtr _sensor, sdf::ElementPtr _
   // Get tf frame for output
   impl_->frame_name_ = gazebo_ros::SensorFrameID(*_sensor, *_sdf);
 
-  // Image publisher
-  // TODO(louise) Migrate image_connect logic once SubscriberStatusCallback is ported to ROS2
-  impl_->image_pub_ = image_transport::create_publisher(impl_->ros_node_.get(),
-      impl_->camera_name_ + "/image_raw");
+  if (impl_->sensor_type_ != GazeboRosCameraPrivate::MULTICAMERA) {
+    // Image publisher
+    // TODO(louise) Migrate image_connect logic once SubscriberStatusCallback is ported to ROS2
+    impl_->image_pub_.push_back(image_transport::create_publisher(impl_->ros_node_.get(),
+      impl_->camera_name_ + "/image_raw"));
 
-  // TODO(louise) Uncomment this once image_transport::Publisher has a function to return the
-  // full topic.
-  // RCLCPP_INFO(impl_->ros_node_->get_logger(), "Publishing images to [%s]",
-  //  impl_->image_pub_.getTopic());
+    // TODO(louise) Uncomment this once image_transport::Publisher has a function to return the
+    // full topic.
+    // RCLCPP_INFO(impl_->ros_node_->get_logger(), "Publishing images to [%s]",
+    //  impl_->image_pub_.getTopic());
 
-  // Camera info publisher
-  // TODO(louise) Migrate ImageConnect logic once SubscriberStatusCallback is ported to ROS2
-  impl_->camera_info_pub_ = impl_->ros_node_->create_publisher<sensor_msgs::msg::CameraInfo>(
-    impl_->camera_name_ + "/camera_info", rclcpp::SensorDataQoS());
+    // Camera info publisher
+    // TODO(louise) Migrate ImageConnect logic once SubscriberStatusCallback is ported to ROS2
+    impl_->camera_info_pub_.push_back(
+      impl_->ros_node_->create_publisher<sensor_msgs::msg::CameraInfo>(
+        impl_->camera_name_ + "/camera_info",
+        rclcpp::SensorDataQoS()));
 
-  RCLCPP_INFO(impl_->ros_node_->get_logger(), "Publishing camera info to [%s]",
-    impl_->camera_info_pub_->get_topic_name());
+    RCLCPP_INFO(impl_->ros_node_->get_logger(), "Publishing camera info to [%s]",
+      impl_->camera_info_pub_.back()->get_topic_name());
+
+  } else {
+    for (uint64_t i = 0; i < impl_->num_cameras_; ++i) {
+      // Image publisher
+      impl_->image_pub_.push_back(image_transport::create_publisher(impl_->ros_node_.get(),
+        impl_->camera_name_ + "/" + gazebo::MultiCameraPlugin::camera[i]->Name() + "/image_raw"));
+
+      // RCLCPP_INFO(impl_->ros_node_->get_logger(), "Publishing %s camera images to [%s]",
+      //   gazebo::MultiCameraPlugin::camera[i]->Name().c_str(),
+      //   impl_->image_pub_.back().getTopic());
```

Comment:
You already have this below, you can remove the comment

Reply:
The `RCLCPP_INFO` below this https://github.com/ros-simulation/gazebo_ros_pkgs/blob/c377a9632f92a7f5a9fd34ec78399c2b89ffb2e1/gazebo_plugins/src/gazebo_ros_camera.cpp#L214-L216 is for the camera info. This one is for the images.